jetty-core/jetty-websocket/jetty-websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/WebSocketCoreSession.java

```diff
@@ -355,7 +355,7 @@ public void processHandlerError(Throwable cause, Callback callback)
             code = ((CloseException)cause).getStatusCode();
         else if (cause instanceof ClosedChannelException)
             code = CloseStatus.NO_CLOSE;
-        else if (cause instanceof Utf8Appendable.NotUtf8Exception)
+        else if (cause instanceof CharacterCodingException)
```

Comment:
Just get rid of this if case entirely.
The new code throwing `BadPayloadException` is enough, as that class extends from `CloseException` already.